PSReadLine/Prediction.Entry.cs

```diff
@@ -0,0 +1,351 @@
+                                break;
+                            }
+
+                            int midRightLenInCells = LengthInBufferCells(SuggestionText, InputMatchIndex, SuggestionText.Length);
+                            if (midRightLenInCells <= threshold)
+                            {
+                                // Otherwise, if the (mid+right) portions take up to 2/3 of the text width, we just truncate the suggestion text at the beginning.
+                                int leftStrLen = SubstringLengthByCellsFromEnd(SuggestionText, InputMatchIndex - 1, textWidth - midRightLenInCells - 3);
+                                line.Append("...")
+                                    .Append(SuggestionText, InputMatchIndex - leftStrLen, leftStrLen)
+                                    .Append(_singleton._options.EmphasisColor)
+                                    .Append(SuggestionText, InputMatchIndex, input.Length)
+                                    .EndColorSection(selectionHighlighting)
+                                    .Append(SuggestionText, rightStartindex, SuggestionText.Length - rightStartindex);
+                                break;
+                            }
+
+                            int midLenInCells = LengthInBufferCells(SuggestionText, InputMatchIndex, InputMatchIndex + input.Length);
+                            threshold = DivideAndRoundUp(textWidth, 3);
+                            if (midLenInCells <= threshold)
+                            {
+                                // Otherwise, if the mid portion takes up to 1/3 of the text width, we truncate the suggestion text
+                                // at both the beginning and the end.
+                                int leftCellLen = (textWidth - midLenInCells) / 2;
+                                int rigthCellLen = textWidth - midLenInCells - leftCellLen;
+
+                                int leftStrLen = SubstringLengthByCellsFromEnd(SuggestionText, InputMatchIndex - 1, leftCellLen - 3);
+                                int rightStrLen = SubstringLengthByCells(SuggestionText, rightStartindex, rigthCellLen - 3);
+
+                                line.Append("...")
+                                    .Append(SuggestionText, InputMatchIndex - leftStrLen, leftStrLen)
+                                    .Append(_singleton._options.EmphasisColor)
+                                    .Append(SuggestionText, InputMatchIndex, input.Length)
+                                    .EndColorSection(selectionHighlighting)
+                                    .Append(SuggestionText, rightStartindex, rightStrLen)
+                                    .Append("...");
+                                break;
+                            }
+
+                            int leftPlusRightLenInCells = leftMidLenInCells + midRightLenInCells - midLenInCells * 2;
+                            if (leftPlusRightLenInCells <= textWidth - 7)
```

Comment:
Seems like `7` should be a const defined somewhere

Reply:
`7` is completely a guessing number, and it's explained in the comment right below it:
> ... if the (left+right) portions are not too long -- namely we can reserve 7 cells at least for the mid portion, ...

So I think I will just keep it as is without defining a const for it.